libs/net/dal/Services/ReportService.cs

```diff
@@ -220,9 +227,23 @@ public override Report Update(Report entity)
     public async Task<Dictionary<string, Elastic.Models.SearchResultModel<API.Areas.Services.Models.Content.ContentModel>>> FindContentWithElasticsearchAsync(string index, Report report)
     {
         var results = new Dictionary<string, Elastic.Models.SearchResultModel<API.Areas.Services.Models.Content.ContentModel>>();
+        var reportSettings = JsonSerializer.Deserialize<ReportSettingsModel>(report.Settings.ToJson(), _serializerOptions) ?? new();
```

Comment:
Moved a lot of logic into this function to improve performance and complexity of the different code that uses it.